distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/SystemJvmOptions.java

```diff
@@ -12,18 +12,20 @@
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.EsExecutors;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.util.List;
 import java.util.Map;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 final class SystemJvmOptions {
 
     static List<String> systemJvmOptions(Settings nodeSettings, final Map<String, String> sysprops) {
         String distroType = sysprops.get("es.distribution.type");
         boolean isHotspot = sysprops.getOrDefault("sun.management.compiler", "").contains("HotSpot");
-
-        return Stream.concat(
+        boolean useEntitlements = Boolean.parseBoolean(sysprops.getOrDefault("es.entitlements.enabled", "false"));
+        return Stream.of(
```

Comment:
I presume at some point we'll want to stop passing `-Djava.security.manager=allow` when this is enabled. I guess it's benign to keep it for now.

Reply:
That's a good idea. I think this PR could add that.

Reply:
I've looked into it a little and that would be a nontrivial change. There are at least two other places affected:
- `initPhase1` installs a dummy `SecurityManager`
- `AllPermissionsCheck` checks that the `SecurityManager` is working

Replacing these is not difficult but it's not trivial either. Perhaps this can wait for another PR?

Reply:
💯 No need to address in this PR. That said, we'll need to address it before we start testing on Java 24 as I presume it'll blow up if we try to pass this.